packages/orm/src/relations/ManyToOneReference.ts

```diff
@@ -149,40 +162,42 @@ export class ManyToOneReference<T extends Entity, U extends Entity, N extends ne
         o2m.set(undefined as any);
       }
     }
-    setField(this.entity, this.fieldName as string, undefined);
+    setField(this.entity, this.fieldName, undefined);
     this.loaded = undefined as any;
     this._isLoaded = true;
   }
 
   // Internal method used by OneToManyCollection
   setImpl(other: U | N): void {
+    // If other is new (i.e. has no id), we only noop/early exit if it matches our loaded reference.
+    // Otherwise, noop/early exit based on id comparison (b/c we may not be loaded yet).
     if (other?.isNewEntity ? other === this.loaded : this.id === other?.id) {
       return;
     }
-
-    // we may not be loaded yet, but our previous entity might already be in the UoW
-    const previousLoaded = this.loaded ?? this.maybeFindExisting();
```

Comment:
Well shoot, I thought there was a bug here that `previousLoaded` was, on `main`, getting set only _after_ the field had already changed, but it looks like `main` was correct, and I just had some mis-refactored code on my `aggregate-roots` branch.

Hrm. Well, I think the code is still cleaner now, and we learned to have a `ManyToOneReference.id` setter, so going to keep the PR, but I guess not technically a bug fix.